tests/unit/test_table_options.py

```diff
@@ -104,111 +104,151 @@ def test_table_clustering_fields_dialect_option_type_error(faux_conn):
+    """
+
     with pytest.raises(sqlite3.OperationalError):
         setup_table(
             faux_conn,
             "some_table",
             sqlalchemy.Column("id", sqlalchemy.Integer),
-            sqlalchemy.Column("createdAt", sqlalchemy.TIMESTAMP),
+            sqlalchemy.Column("createdAt", sqlalchemy.DateTime),
             bigquery_time_partitioning=TimePartitioning(field="createdAt"),
         )
-
-    assert " ".join(faux_conn.test_data["execute"][-1][0].strip().split()) == (
-        "CREATE TABLE `some_table` ( `id` INT64, `createdAt` TIMESTAMP )"
-        " PARTITION BY TIMESTAMP_TRUNC(createdAt, DAY)"
+    result = " ".join(faux_conn.test_data["execute"][-1][0].strip().split())
+    expected = (
+        "CREATE TABLE `some_table` ( `id` INT64, `createdAt` DATETIME )"
+        " PARTITION BY DATETIME_TRUNC(createdAt, DAY)"
     )
+    assert result == expected
 
 
-def test_table_time_partitioning_with_date_dialect_option(faux_conn):
-    # expect table creation to fail as SQLite does not support partitioned tables
+@pytest.mark.parametrize(
+    "column_dtype,time_partitioning_type",
+    [
+        pytest.param(
+            sqlalchemy.DATE,
+            TimePartitioningType.HOUR,
+            marks=pytest.mark.xfail,
+        ),
+        (sqlalchemy.DATE, TimePartitioningType.DAY),
+        (sqlalchemy.DATE, TimePartitioningType.MONTH),
+        (sqlalchemy.DATE, TimePartitioningType.YEAR),
+    ],
+)
+def test_table_time_partitioning_with_partitiondate_option(
+    faux_conn,
+    column_dtype,
+    time_partitioning_type,
+):
+    """NOTE: Expect table creation to fail as SQLite does not support
+    partitioned tables, despite that, we are still able to test the generation
+    of SQL statements
+
+    If the `field` argument to TimePartitioning() is not provided, it detaults to
+    None. That causes the pseudocolumn "_PARTITIONDATE" to be used by default as
+    the column to partition by.
+
+    _PARTITIONTIME only returns a result if TimePartitioningType is DAY, MONTH,
+    YEAR. BigQuery cannot partition on _PARTITIONDATE by HOUR, so that is
+    expected to xfail.
+    """
+
     with pytest.raises(sqlite3.OperationalError):
         setup_table(
             faux_conn,
             "some_table_2",
             sqlalchemy.Column("id", sqlalchemy.Integer),
-            sqlalchemy.Column("createdAt", sqlalchemy.DATE),
-            bigquery_time_partitioning=TimePartitioning(field="createdAt"),
+            sqlalchemy.Column("createdAt", column_dtype),
+            bigquery_time_partitioning=TimePartitioning(type_=time_partitioning_type),
         )
 
     # confirm that the following code creates the correct SQL string
-    assert " ".join(faux_conn.test_data["execute"][-1][0].strip().split()) == (
-        "CREATE TABLE `some_table_2` ( `id` INT64, `createdAt` DATE )"
-        " PARTITION BY createdAt"
+    result = " ".join(faux_conn.test_data["execute"][-1][0].strip().split())
```

Comment:
Same as https://github.com/googleapis/python-bigquery-sqlalchemy/pull/1113/files#r1852967358

Reply:
In this test, we do NOT provide a `field` argument to the TimePartitioning class whereas in the tests above, we do.

This is needed to confirm that in the absence of a `field` argument the SQL statement will not use a function, etc but will instead use `_PARTITIONDATE`

Both the tests above end up including a truncation function but this on does not (and should not).